fw/http.c

```diff
@@ -3375,6 +3459,61 @@ tfw_h1_set_loc_hdrs(TfwHttpMsg *hm, bool is_resp, bool from_cache)
 	return 0;
 }
 
+static int
+tfw_h1_add_loc_hdrs(TfwHttpMsg *hm, const TfwHdrMods *h_mods)
+{
+	unsigned int i;
+	TfwHttpHdrTbl *ht = hm->h_tbl;
+
+	if (!h_mods)
+		return 0;
+
+	for (i = 0; i < h_mods->sz; ++i) {
+		const TfwHdrModsDesc *desc = &h_mods->hdrs[i];
+		int r;
+		unsigned short hid = desc->hid;
+		TfwStr h_mdf = {
+			.chunks = (TfwStr []){
+				{},
+				{ .data = S_DLM, .len = SLEN(S_DLM) },
+				{}
+			},
+			.len = SLEN(S_DLM),
+			.nchunks = 2 /* header name + delimeter. */
+		};
+
+		/* FIXME: this is a temporary WA for GCC12, see #1695 for details */
+		if (TFW_STR_CHUNK(desc->hdr, 1) == NULL)
+			continue;
+
+		if (unlikely(desc->append && !TFW_STR_EMPTY(&ht->tbl[hid])
+			     && (hid < TFW_HTTP_HDR_NONSINGULAR)))
+		{
+			T_WARN("Attempt to add already existed singular header '%.*s'\n",
+			PR_TFW_STR(TFW_STR_CHUNK(desc->hdr, 0)));
+			continue;
```

Comment:
Should we continue with error message here or return error?

Reply:
Yes we should, it is not error, just warning for user